includes/compat/trait-wc-stripe-subscriptions.php

```diff
@@ -363,6 +363,21 @@ public function process_subscription_payment( $amount, $renewal_order, $retry =
 				throw new WC_Stripe_Exception( print_r( $response, true ), $localized_message );
 			}
 
+			// TODO: Remove when SEPA is migrated to payment intents.
+			if ( 'stripe_sepa' !== $this->id ) {
+				$this->unlock_order_payment( $renewal_order );
+			}
+		} catch ( WC_Stripe_Exception $e ) {
+			WC_Stripe_Logger::log( 'Error: ' . $e->getMessage() );
+
+			do_action( 'wc_gateway_stripe_process_payment_error', $e, $renewal_order );
+
+			/* translators: error message */
+			$renewal_order->update_status( 'failed' );
+		}
+
+		try {
+
```

Comment:
The diff makes it look like there are some new changes. But the main change here is I have moved this [block](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/includes/compat/trait-wc-stripe-subscriptions.php#L366-L405) to another try-catch below, where catch block logs the error only.